src/examples/zkapps/dex/run.ts

```diff
@@ -36,160 +31,487 @@ console.log('TOKEN X ID\t', Token.Id.toBase58(tokenIds.X));
+   *   token account (in parent tokens, which is Mina) and transfers amount of minted lqXY tokens to user’s account;
+   */
+  expect(balances.user.X).toEqual(oldBalances.user.X - USER_DX);
+  expect(balances.user.Y).toEqual(
+    oldBalances.user.Y - (USER_DX * oldBalances.dex.Y) / oldBalances.dex.X
+  );
+  expect(balances.user.MINA).toEqual(oldBalances.user.MINA - 1n);
+  expect(balances.user.lqXY).toEqual(
+    (USER_DX * oldBalances.total.lqXY) / oldBalances.dex.X
+  );
+
+  /**
+   * Happy path (lqXY token exists for users account)
+   *
+   * Same case but we are checking that no token creation fee is paid by the liquidity supplier.
+   *
+   * Note: with vesting, this is a failure case because we can't change timing on an account that currently has an active timing
+   */
+  USER_DX = 1000n;
+  console.log('user supply liquidity (2)');
+  tx = await Mina.transaction(keys.user, () => {
+    dex.supplyLiquidity(addresses.user, UInt64.from(USER_DX));
+  });
+  await tx.prove();
+  tx.sign([keys.user]);
+  if (!withVesting) {
+    await tx.send();
+    [oldBalances, balances] = [balances, getTokenBalances()];
+    console.log('DEX liquidity (X, Y):', balances.dex.X, balances.dex.Y);
+    console.log('user DEX tokens:', balances.user.lqXY);
+    expect(balances.user.X).toEqual(oldBalances.user.X - USER_DX);
+    expect(balances.user.Y).toEqual(
+      oldBalances.user.Y - (USER_DX * oldBalances.dex.Y) / oldBalances.dex.X
+    );
+    expect(balances.user.MINA).toEqual(oldBalances.user.MINA);
+    expect(balances.user.lqXY).toEqual(
+      oldBalances.user.lqXY +
+        (USER_DX * oldBalances.total.lqXY) / oldBalances.dex.X
+    );
+  } else {
+    await expect(tx.send()).rejects.toThrow(
+      /Update_not_permitted_timing_existing_account/
+    );
+  }
+
+  /**
+   * Check the method failures during an attempts to supply liquidity when:
+   * - There is no token X or Y (or both) created yet for user’s account;
+   * - There is not enough tokens available for user’s tokens accounts, one is willing to supply;
+   */
+  console.log('supplying with no tokens (should fail)');
+  tx = await Mina.transaction(keys.user2, () => {
+    AccountUpdate.fundNewAccount(keys.user2);
+    dex.supplyLiquidityBase(
+      addresses.user2,
+      UInt64.from(100),
+      UInt64.from(100)
+    );
+  });
+  await tx.prove();
+  tx.sign([keys.user2]);
+  await expect(tx.send()).rejects.toThrow(/Overflow/);
+  console.log('supplying with insufficient tokens (should fail)');
+  tx = await Mina.transaction(keys.user, () => {
+    dex.supplyLiquidityBase(addresses.user, UInt64.from(1e9), UInt64.from(1e9));
+  });
+  await tx.prove();
+  tx.sign([keys.user]);
+  await expect(tx.send()).rejects.toThrow(/Overflow/);
+
+  /**
+   * - Resulting operation will overflow the SC’s receiving token by type or by any other applicable limits;
+   *
+   * note: this throws not at the protocol level, but because the smart contract multiplies two UInt64s which overflow.
```

Comment:
what do you mean here? There is a constraint failure due to the overflow, right? It's just due to the smart contract.

Reply:
Yes there is a constraint failure. So we don't get to the point where we could trigger an overflow failure in the protocol logic. In that sense, this is not a proper test as it was intended